libraries/provider_configure.rb

```diff
@@ -0,0 +1,91 @@
+class Chef
+  # Chef Provider for configuring an elasticsearch instance
+  class Provider::ElasticsearchConfigure < Provider
+    include Poise
+
+    def action_manage
+      # Create ES directories
+      #
+      [ new_resource.path_conf, new_resource.path_logs ].each do |path|
+        directory path do
+          owner new_resource.user
+          group new_resource.group
+          mode 0755
+          recursive true
+          action :create
+        end
+      end
+
+      # Create data path directories
+      #
+      data_paths = new_resource.path_data.is_a?(Array) ? new_resource.path_data : new_resource.path_data.split(',')
+
```

Comment:
I think the chef pattern would be to reuse the directory resource where-ever possible, until we encounter features we need but the directory provider in chef doesn't have (and even then, we could call the provider in Ruby, instead of the recipe DSL). But the common chef pattern would also be to split out the logging configuration into a separate resource and provider. I stopped at this design, but I think it's probably better to split it out.

Basically anything that feels like a separate 'object' in the system should be it's own resource/provider. So probably we could take the logging file and move it out of the configure resource/provider. This would have the added benefit of making it optional.